.github/workflows/haskell-ci.yml

```diff
@@ -26,7 +27,7 @@ jobs:
           version: "0.17.0.0"
   linux:
     name: Haskell-CI - Linux - ${{ matrix.compiler }}
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```

Comment:
nit: I prefer to keep this on latest do we can fail forward. Understandably this is autogenerated so you can keep it as is.